src/calculators/LC_LJCalculator.cpp

```diff
@@ -0,0 +1,214 @@
+        calculateLC_X(linkedCellContainer, delta_t);
+        calculateLC_V(linkedCellContainer, delta_t);
+        linkedCellContainer.updateCells();
+    }
+
+    void LC_LJCalculator::calculateGravityLC(LinkedCellContainer &linkedCellContainer) {
+        for (auto &p: linkedCellContainer) {
+            const double grav_force = p->getM() * this->g_grav;
+            const std::array<double, 3> newForce{p->getF()[0], p->getF()[1] + grav_force, p->getF()[2]};
+            SPDLOG_DEBUG("Applying gravity {}. Current F of Particle at {} is ({}, {}, {})",
+                         grav_force,
+                         ArrayUtils::to_string(p->getX()),
+                         p->getF()[0], p->getF()[1], p->getF()[2]);
+            p->setF(newForce);
+            SPDLOG_DEBUG("Updated F of Particle at {} to ({}, {}, {})",
+                         ArrayUtils::to_string(p->getX()),
+                         p->getF()[0], p->getF()[1], p->getF()[2]);
+        }
+    }
+
+    void LC_LJCalculator::calculateLC_F(LinkedCellContainer &linkedCellContainer) {
+        for (auto &p: linkedCellContainer) {
+            p->setOldF(p->getF());  // Update oldF with currentF
+            p->setF({0, 0, 0});     // Reset F to zeros
+            SPDLOG_DEBUG("Reset force of Particle at {} to ({}, {}, {})", ArrayUtils::to_string(p->getX()), 0, 0, 0);
+            SPDLOG_DEBUG("Updated old force of Particle at {} to {}",
+                         ArrayUtils::to_string(p->getX()),
+                         ArrayUtils::to_string(p->getOldF()));
+        }
+
+        for (auto &particle1: linkedCellContainer) {
+            std::array<double, 3> x1 = particle1->getX();
+            SPDLOG_DEBUG("Processing Particle at position ({}, {}, {})", x1[0], x1[1], x1[2]);
+            std::vector<std::shared_ptr<Cell>> &neighboringCells = linkedCellContainer.getNeighboringCellsIncludingSelf(*particle1);
+            for (auto &cell : neighboringCells) {
+                 SPDLOG_DEBUG("Processing Cell at index ({}, {}, {})", cell->getIndex()[0], cell->getIndex()[1],
+                           cell->getIndex()[2]);
+                for (auto &particle2 : cell->getParticles()) {
+                    if (particle1 == particle2 || particle1->getID() > particle2->getID()) {
+                        continue; // Skip the force calculation for the same particle or duplicate pair
+                    }
+
+                    // LJ force calculation
+                    const std::array<double, 3> x2 = particle2->getX();
+
+                    // Calculate the distance vector and its norm
+                    const std::array<double, 3> dx = ArrayUtils::elementWisePairOp(x1, x2, std::minus<>());
+                    const double distance = ArrayUtils::L2Norm(dx);
+
+                    double epsilon = pow((particle1->getEpsilon() * particle2->getEpsilon()), 0.5);
+                    double sigma = 0.5 * (particle1->getSigma() + particle2->getSigma());
```

Comment:
possible optimization: instead of calculating these values for each pair of particles, you can precompute these for possible pairs of particle types (distinguished using ids maybe?) and use them as `constexpr`. This will reduce a lot of computation I think.